test/connectors/CacheConnectorBehavious.scala

```diff
@@ -0,0 +1,472 @@
+
+        connector().upsert("foo", json) map {
+          _ mustEqual json
+        }
+      }
+
+      "add fields to existing data" in {
+
+        val json = Json.obj(
+          "foo" -> "bar"
+        )
+
+        val updatedJson = Json.obj(
+          "foo" -> "bar",
+          "fake-identifier" -> "foobar"
+        )
+
+        val value = Json.stringify(json)
+        val updatedValue = Json.stringify(updatedJson)
+
+        server.stubFor(
+          get(urlEqualTo(url("foo")))
+            .willReturn(
+              ok(value)
+            )
+        )
+
+        server.stubFor(
+          post(urlEqualTo(url("foo")))
+            .withRequestBody(equalTo(updatedValue))
+            .willReturn(
+              ok
+            )
+        )
+
+        connector().upsert("foo", updatedJson) map {
+          _ mustEqual updatedJson
+        }
+      }
+
+      "update existing data" in {
+
+        val json = Json.obj(
+          "fake-identifier" -> "foo"
+        )
+
+        val updatedJson = Json.obj(
+          "fake-identifier" -> "foobar"
+        )
+
+        val value = Json.stringify(json)
+        val updatedValue = Json.stringify(updatedJson)
+
+        server.stubFor(
+          get(urlEqualTo(url("foo")))
+            .willReturn(
+              ok(value)
+            )
+        )
+
+        server.stubFor(
+          post(urlEqualTo(url("foo")))
+            .withRequestBody(equalTo(updatedValue))
+            .willReturn(
+              ok
+            )
+        )
+
+        connector().upsert("foo", updatedJson) map {
+          _ mustEqual updatedJson
+        }
+      }
+
+      "return a failed future on upstream error" in {
+
+        val json = Json.obj(
+          "fake-identifier" -> "foo"
+        )
+
+        val updatedJson = Json.obj(
+          "fake-identifier" -> "foobar"
+        )
+
+        val value = Json.stringify(json)
+        val updatedValue = Json.stringify(updatedJson)
+
+        server.stubFor(
+          get(urlEqualTo(url("foo")))
+            .willReturn(
+              ok(value)
+            )
+        )
+
+        server.stubFor(
+          post(urlEqualTo(url("foo")))
+            .withRequestBody(equalTo(updatedValue))
+            .willReturn(
+              serverError
+            )
+        )
+
+
+        recoverToExceptionIf[HttpException] {
+          connector().upsert("foo", updatedJson)
+        } map {
+          _.responseCode mustEqual Status.INTERNAL_SERVER_ERROR
+        }
+
+      }
+    }
+
+    ".remove" must {
+      "remove existing data" in {
+        val json = Json.obj(
+          FakeIdentifier.toString -> "fake value",
+          "other-key" -> "meh"
+        )
+
+        val updatedJson = Json.obj(
+          "other-key" -> "meh"
+        )
+
+        val value = Json.stringify(json)
+        val updatedValue = Json.stringify(updatedJson)
+
+        server.stubFor(
+          get(urlEqualTo(url("foo")))
+            .willReturn(
+              ok(value)
```

Comment:
File name is spelt wrongly: should be BehaviouRs

Reply:
fixed